nutils/numeric.py

```diff
@@ -439,6 +439,61 @@ def binom(n, k):
     b *= i
   return a // b
 
+def full(shape, fill_value):
+  return numpy.lib.stride_tricks.as_strided(fill_value, shape, [0]*len(shape), writeable=False)
+
+def writeable(array):
+  '''Test if any base is writeable.'''
+
+  return array.flags.writeable or isinstance(array.base, numpy.ndarray) and writeable(array.base)
+
+def freeze(array):
```

Comment:
Should this also deal with views created by `stride_tricks` as in `types._deepfreeze`? And if so should this function be called there?

Reply:
Yeah I'm struggling a bit with choosing the right modules, but my intention is indeed to keep only deepfreeze. I'm now thinking of moving `Py_buffer` to util and `keepfrozen` to types.